iree/turbine/kernel/compiler/dispatch_codegen.py

```diff
@@ -193,16 +200,22 @@ def abi_type(binding: BindingDesc):
             # TODO: Support passing workload to the dispatch function.
             from ..wave.codegen import gen_sympy_index
 
+            arguments = workgroup_builder.entry_block.arguments
+
             # Map dynamic symbols to block arguments.
             dynamic_symbols_mapping = {
-                k: v
-                for k, v in zip(
-                    dynamic_symbols, workgroup_builder.entry_block.arguments
-                )
+                k.symbol_type: v for k, v in zip(dynamic_dim_bindings, arguments)
             }
 
             with InsertionPoint(workgroup_builder.entry_block):
-                result_type = IndexType.get()
+                for i, s in enumerate(scalar_bindings):
+                    if s.symbol_type is None:
+                        continue
+
+                    offset = len(dynamic_dim_bindings) + i
+                    dynamic_symbols_mapping[s.symbol_type] = arguments[offset]
```

Comment:
Why not create a separate mapping for scalar_bindings? This seems to be using dynamic_symbols_mapping to hold both dynamic and scalar bindings?

Reply:
It's used in `gen_sympy_index` call in few lines so even if we had a separate mapping we'd need to merge them before the call.

Reply:
okay makes sense, then I would just rename this variable to make it clear that it is not just dynamic symbols.

Reply:
Renamed to `symbols_mapping`, but not sure if it's much better.